File: src/src_platform_android/fmg/android/utils/StaticInitializer.java
package fmg.android.utils;

import android.os.Handler;
import android.os.Looper;
import android.util.Log;

import fmg.android.img.Animator;
import fmg.common.Color;
import fmg.common.LoggerSimple;
import fmg.common.ui.Factory;
import fmg.core.mosaic.MosaicDrawModel;

public final class StaticInitializer {

    static {
        Factory.DEFERR_INVOKER = new Handler(Looper.getMainLooper())::post;
        Factory.GET_ANIMATOR = Animator::getSingleton;
        Factory.TIMER_CREATOR = Timer::new;

        MosaicDrawModel.DefaultBkColor = new Color(0xFFEEEEEE); // #EEEEEE or #FAFAFA

        LoggerSimple.DEFAULT_WRITER = message -> Log.d("fmg", message);
    }

    public static void init() {
        // implicit call static block
    }

}

File: src/src_platform_android/fmg/android/utils/StaticInitializer.kt
package fmg.android.utils

import java.util.function.Consumer
import java.util.function.Supplier

import android.os.Handler
import android.os.Looper
import android.util.Log

import fmg.common.Color
import fmg.common.LoggerSimple
import fmg.common.ui.ITimer
import fmg.common.ui.Factory
import fmg.core.img.IAnimator
import fmg.core.mosaic.MosaicDrawModel
import fmg.android.img.Animator

object StaticInitializer {

    init {
        Factory.DEFERR_INVOKER = Consumer<Runnable> { Handler(Looper.getMainLooper()).post(it) }
        Factory.GET_ANIMATOR = Supplier<IAnimator> { Animator.singleton }
        Factory.TIMER_CREATOR = Supplier<ITimer> { Timer() }

        MosaicDrawModel.DefaultBkColor = Color(-0x111112) // #EEEEEE or #FAFAFA

        LoggerSimple.DEFAULT_WRITER = Consumer<String> { Log.d("fmg", it) }
    }

    fun init() {
        // implicit call static block
    }

}
